core/src/addresses.ts

```diff
@@ -0,0 +1,68 @@
+import type { Contract } from 'ethers';
+import { ethers } from 'ethers';
+import memoizee from 'memoizee';
+import COLLATERALS, { getAllCollateralTypes } from './constants/COLLATERALS';
+import getProvider from './provider';
+import CHAINLOG from './abis/CHAINLOG.json';
+import { CHAINLOG_ADDRESS } from './constants/NETWORKS';
+
+const CHAINLOG_CACHE = 24 * 60 * 60 * 1000;
```

Comment:
> CHAINLOG_CACHE

Maybe instead `CHAINLOG_CACHE_EXPIRY_MS`?

Also, let's make this configurable, it will be helpful to avoid memoization for debugging of deployments (:

Reply:
This is the file that was only renamed, not sure why github shows it as a completely new one. I suggest to do this change in some other refactoring/renaming PR.